tests/integrations/base_tester.py

```diff
@@ -0,0 +1,280 @@
+the commands (i.e. train, export, deploy).
+
+- {Integration_name}_tester.py file containing integration testing class inherited
+from BaseIntegrationTester
+
+- Tests to run after commands are run. Tests should be implemented in the tester class
+described in B and should be decorated by @skip_inactive_stage found in helpers.py
+
+- .yaml file for each scenario to run
+"""
+
+import os
+import subprocess
+from time import sleep
+from typing import Dict, List, Union
+
+import pytest
+import yaml
+from pydantic import BaseModel
+
+from tests.integrations.helpers import stream_process
+
+
+class BaseIntegrationTester:
+    """
+    Base class for testing integrations through train, export, and deploy scripts.
+
+    Each of train, export, and deploy are "command types" and can constitute a stage
+    in a multi-stage run.
+
+    Each integration will implement a subclass of this class, with the following
+    fields and functions filled out:
+
+    :field command_stubs: Mapping from command type to the respective CLI command.
+        Note that the stubs can be modified via the get_root_commands() function
+    :field command_args_classes: Mapping from command type to the pydnatic class
+        which holds the CLI args for that command
+    :function teardown: Perform the appropriate post-testing teardown, including
+        file cleanup
+    :function get_root_commands: [Optional] If the CLI root commands are dynamic
+        (e.g. transformers commands are task-dependent), override this function to
+        return the correct stubs
+    :function capture_pre_run_state: [Optional] Used to save any information about
+        the pre-run state which may be needed for testing post-run
+    :function check_teardown: [Optional] Add checks for a successful environment
+        cleanup
+    :function save_stage_information [Optional] Save state information in between
+        stage runs
+    """
+
+    command_stubs = {
+        "train": "sparseml.foo_integration.train",
+        "export": "sparseml.foo_integration.export",
+        "deploy": "sparseml.foo_integration.deploy",
+    }
+    command_args_classes = {
+        "train": BaseModel,
+        "export": BaseModel,
+        "deploy": BaseModel,
+    }
+
+    @pytest.fixture(
+        scope="class",
+        autouse="true",
+    )
+    def setup(self, request):
+        self = request.cls
+
+        config_path = request.param
+        with open(config_path) as f:
+            raw_config = yaml.safe_load(f)
+
+        # Remove cadence for easier processing. It's saved, but not utilized later
+        self.cadence = raw_config.pop("cadence")
+
+        # Command types present in this config
+        self.command_types = [_type for _type in raw_config]
+
+        # Final command stub
+        self.command_stubs_final = self.get_root_commands(raw_config)
+
+        # Map each present command type to the pydantic class and pre-args (see below)
+        self.configs = {
```

Comment:
thoughts on creating a separate class to handle the loading/parsing of config files? can also include documentation on how to write configs there. happy to take this on as well in an update 🙂